### PR TITLE
Add support for running govulncheck

### PR DESCRIPTION
**Pull request by 51n15t9r**
Add the possibility to run a Go vulnerability check on Prometheus Components

### CODE REVIEW

**Comment by 51n15t9r** [comment]
@simonpasquier - Could you review this, please?

**Comment by roidelapluie** [comment]
Thank you for you pull requests, we are already using codeql and I am wondering what are the advantages of govulncheck and whether we should make the move to it?

**Review by bwplotka — APPROVED** [review]
(During bug scrub)

Nice! Sounds like it's an official Go tool (https://go.dev/blog/govulncheck) and it's not in golangci-lint (https://github.com/golangci/golangci-lint/pull/3199). However CI action exists for it (https://github.com/prometheus/prometheus/pull/12654/files) which means we don't technically need a Makefile cmd for that (this is what we do for golangci-lint too (https://github.com/prometheus/prometheus/blob/main/.github/workflows/ci.yml#L151).

Still we could merge this if you think it would be useful for local (pre PR) checks, WDYT @51n15t9r ?

File: Makefile.common [hunk 1]
```diff
@@ -267,3 +267,9 @@ $(1)_precheck:
 		exit 1; \
 	fi
 endef
+
+govulncheck: install-govulncheck
+	govulncheck ./...
+
+install-govulncheck:
+	which govulncheck || go install golang.org/x/vuln/cmd/govulncheck@latest
```
Comment (bboreham) [inline Makefile.common:275]: prefer `command -v govulncheck > /dev/null` over `which govulncheck`
See https://unix.stackexchange.com/a/85250/67917

**Comment by bwplotka** [comment]
We could also add an issue to add CI job for it or do it in this PR.

File: Makefile.common (hunk 1 above)
Comment (51n15t9r) [reply Makefile.common]: Thank you for the review. I've updated my changes as per the recommendation.

**Comment by 51n15t9r** [comment]
> (During bug scrub)
> 
> Nice! Sounds like it's an official Go tool (https://go.dev/blog/govulncheck) and it's not in golangci-lint ([golangci/golangci-lint#3199](https://github.com/golangci/golangci-lint/pull/3199)). However CI action exists for it (https://github.com/prometheus/prometheus/pull/12654/files) which means we don't technically need a Makefile cmd for that (this is what we do for golangci-lint too (https://github.com/prometheus/prometheus/blob/main/.github/workflows/ci.yml#L151).
> 
> Still we could merge this if you think it would be useful for local (pre PR) checks, WDYT @51n15t9r ?

Yes - That was my intention. To be able to run a check pre PR.

**Comment by 51n15t9r** [comment]
> We could also add an issue to add CI job for it or do it in this PR.

If this change is okay to be merged, I can push the CI changes in a separate PR.

**Review by aknuds1 — APPROVED** [review]
LGTM.

Do you have any comment though regarding @roidelapluie's question about the advantages versus CodeQL, and whether we should make the move to govulncheck instead of CodeQL? Are the two perhaps complementary?

Also, could you rebase on latest main, so we can try to pass the CI checks?

**Comment by 51n15t9r** [comment]
> LGTM.
> 
> Do you have any comment though regarding @roidelapluie's question about the advantages versus CodeQL, and whether we should make the move to govulncheck instead of CodeQL? Are the two perhaps complementary?
> 
> Also, could you rebase on latest main, so we can try to pass the CI checks?

Thanks @aknuds1 

Code has been rebased. 

While I'm unfamiliar with CodeQL, it seems like a general-purpose code analysis tool, probably not ideal for pre-commit checks that should be used by developers.

govulncheck, on the other hand, could be a great add-on. It is simple to use and and is officially maintained by the Go team. While it does have [limitations](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck#hdr-Limitations), it can be useful for initial checks during go dependency changes

In my opinion, both tools can complement each other.  govulncheck can be a developer's first check when managing dependencies, while CodeQL provides a more comprehensive analysis.

**Comment by aknuds1** [comment]
Sorry, forgot about this in the meantime :/

**Pull request merged**